pythran/pythonic/python/core.hpp

```diff
@@ -25,6 +25,15 @@ auto to_python(T &&value) -> decltype(pythonic::to_python<
       typename std::remove_cv<typename std::remove_reference<T>::type>::type>::
       convert(std::forward<T>(value));
 }
+
+template <class T>
+auto to_python_from_expr(T &&value) -> decltype(to_python(
+      typename pythonic::returnable<typename std::remove_cv<typename std::remove_reference<T>::type>::type>::type{std::forward<T>(value)}))
+{
+  using returnable_type = typename pythonic::returnable<typename std::remove_cv<typename std::remove_reference<T>::type>::type>::type;
+  return to_python(returnable_type{std::forward<T>(value)});
+}
+
```

Comment:
expr is really specific to ndarray. I guess it should be with `to_python<types::numpy_iexpr<Arg>>` in ndarray.hpp or do you have another usage?